railties/lib/rails/generators/rails/app/app_generator.rb

```diff
@@ -563,7 +563,12 @@ def prepare!
       end
 
       def self.default_rc_file
-        File.expand_path("~/.railsrc")
+        xdg_railsrc = File.expand_path("rails/railsrc", ENV["XDG_CONFIG_HOME"]) if ENV["XDG_CONFIG_HOME"]
+        if xdg_railsrc && File.exist?(xdg_railsrc)
```

Comment:
According to the XDG Base Directory Specification, if `$XDG_CONFIG_HOME` is either not set or empty, you should use `$HOME/.config`.

> $XDG_CONFIG_HOME defines the base directory relative to which user specific configuration files should be stored. If $XDG_CONFIG_HOME is either not set or empty, a default equal to $HOME/.config should be used.
>
> ref: https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html

It would probably work with the following code.

```ruby
xdg_config_home = ENV["XDG_CONFIG_HOME"].presence || '~/.config'
xdg_railsrc = File.expand_path("rails/railsrc", xdg_config_home)
if File.exist?(xdg_railsrc)
  xdg_railsrc
else
  File.expand_path("~/.railsrc")
end
```

Reply:
@sinsoku I think you are correct in terms of the spec.  The downside of falling back to `~/.config` is that it makes this feature appear to be Linux-specific (even though it isn't), which makes it less likely to be merged.

On the other hand, both [Ruby](https://github.com/ruby/ruby/blob/b7e1de117e20911ed86b229a4c2f912e469c6472/lib/rubygems/defaults.rb#L128) and [git](https://git-scm.com/docs/git-config#Documentation/git-config.txt-XDGCONFIGHOMEgitconfig) fall back to `~/.config` regardless of OS, so I think there's an argument to be made for doing the same.